packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -147,12 +201,36 @@ class _TextSelectionToolbar extends StatelessWidget {
       onPressed: onPressed,
     );
   }
+
+  /// Update the exact position of the triangle
+  void updateTriangleOffsetX(double triangleOffsetX) {
+    _triangleOffsetX = triangleOffsetX;
+    _cancelTimer();
+
+    /// If the setState method is called immediately,
+    /// the toolbar display position is wrong.
+    _timer = Timer(const Duration(milliseconds: 50), () {
```

Comment:
Can you explain this more?  I worry about adding a timer like this.

Reply:
This timer workaround looks like a hack to me.
Maybe the toolbar should be made a `CustomMultiChildLayout`, so you can have the size of the toolbar ready when you position the triangle?